api/v1beta2/configuration_types.go

```diff
@@ -35,9 +35,11 @@ type ConfigurationSpec struct {
 	// +kubebuilder:pruning:PreserveUnknownFields
 	Variable *runtime.RawExtension `json:"variable,omitempty"`
 
-	// Backend stores the state in a Kubernetes secret with locking done using a Lease resource.
-	// TODO(zzxwill) If a backend exists in HCL/JSON, this can be optional. Currently, if Backend is not set by users, it
-	// still will set by the controller, ignoring the settings in HCL/JSON backend
+	// Backend describes the Terraform backend configuration.
+	// This field is needed if the users use a git repo to provide the hcl files and
+	// want to use their custom Terraform backend (instead of the default kubernetes backend type).
+	// Notice: the content in this field will **override** the backend configuration in the inline hcl code or
```

Comment:
How can it override the backend in the inline hcl or from a git repo?

Reply:
Actually, I want to say that if there are two backend block in the module ( may the custom backend and the backend blaock in the inline code or the git repo), the executor job will throw error. I will update the comment.